modules/rules_erlang/2.3.0/presubmit.yml

```diff
@@ -0,0 +1,64 @@
+matrix:
+  platform:
+  - ubuntu2004
+  - macos
+tasks:
+  verify_targets:
+    name: Verify build targets
+    platform: ${{ platform }}
+    environment:
+      ERLANG_VERSION: "24.3"
+    shell_commands:
+    - curl -O https://raw.githubusercontent.com/kerl/kerl/master/kerl
+    - chmod a+x kerl
+    - ./kerl update releases
+    - ./kerl build ${ERLANG_VERSION}
+    - ./kerl install ${ERLANG_VERSION} ~/kerl/${ERLANG_VERSION}
+    - |
+        ERLANG_HOME="$(realpath ~/kerl/${ERLANG_VERSION})"
+        cat << EOF >> .bazelrc
+          build --@rules_erlang//:erlang_version=${ERLANG_VERSION}
+          build --@rules_erlang//:erlang_home=${ERLANG_HOME}
+        EOF
+    build_flags:
+    - --incompatible_strict_action_env
+    build_targets:
+    - '@rules_erlang//shard_suite:escript'
+
+bcr_test_module:
+  module_path: "shard_suite"
```

Comment:
https://github.com/rabbitmq/rules_erlang/tree/2.3.0/shard_suite
It looks like the `shard_suite` directory isn't actually a test module, which should have it's own MODULE.bazel file and WORKSPACE file. 
It should depend on the main module via a `local_path_override` and test it as an actual external dependency (which means you need to fix some load statements).
Example: https://github.com/fmeum/rules_jni/blob/main/tests/MODULE.bazel

Without this setup, you are essentially building `//...` from the main module, which can be simply achieved by replacing `'@rules_erlang//shard_suite:escript'` with `'@rules_erlang//...'

Reply:
`shard_suite` depends on portions of `rules_erlang` to build and test. However, some other portions of `rules_erlang` (rules that support running sharded test via bazel sharding) use `shard_suite` as tool at runtime to correctly perform the sharding. `shard_suite` is light enough that we want to just bundle it with `rules_erlang`. So, in this case, it sounds to me like we do want to just replace @rules_erlang//shard_suite:escript' with `'@rules_erlang//...'`?

Reply:
Yes, I think in that case it's simpler because the currently setup isn't how bcr_test_module is intended to be used.

Reply:
I have made some changes which I believe bring things to where we wanted. Bonus that this allowed me to fix an issue with one of rules_erlang's dependencies that I had previously missed.